presto-main/src/main/java/io/prestosql/operator/scalar/timestamp/VarcharToTimestampCast.java

```diff
@@ -53,6 +54,11 @@ public static long castToShort(@LiteralParameter("p") long precision, @SqlType("
         catch (IllegalArgumentException e) {
             throw new PrestoException(INVALID_CAST_ARGUMENT, "Value cannot be cast to timestamp: " + value.toStringUtf8(), e);
         }
+        catch (DateTimeException e) {
+            //Leverage highly specific error message from the source exception.
+            throw new PrestoException(INVALID_CAST_ARGUMENT,
+                    String.format("Value cannot be cast to timestamp; %s. ", e.getMessage()), e);
```

Comment:
The exception above includes the value.  Do we lose that here?

Reply:
Good point, it shows you the erroneous part of the value but not the full original value.  Will fix it.

Reply:
@dain - Fixed, it now shows the original value as well as the detailed message indicating what was wrong with it.